tools/integration_tests/read_cache/cache_file_for_range_read_false_test.go

```diff
@@ -62,6 +76,38 @@ func (s *cacheFileForRangeReadFalseTest) TestRangeReadsWithCacheMiss(t *testing.
 	validateFileIsNotCached(testFileName, t)
 }
 
+func (s *cacheFileForRangeReadFalseTest) TestConcurrentReads_ReadIsTreatedNonSequentialAfterFileIsRemovedFromCache(t *testing.T) {
+	var testFileNames [2]string
+	var expectedOutcome [2]*Expected
+	testFileNames[0] = setupFileInTestDir(s.ctx, s.storageClient, testDirName, fileSizeForRangeRead, t)
+	testFileNames[1] = setupFileInTestDir(s.ctx, s.storageClient, testDirName, fileSizeForRangeRead, t)
+
+	var wg sync.WaitGroup
+	for i := 0; i < 2; i++ {
+		wg.Add(1)
+		readFileAsync(t, &wg, testFileNames[i], &expectedOutcome[i])
+	}
+	wg.Wait()
+
+	structuredReadLogs := read_logs.GetStructuredLogsSortedByTimestamp(setup.LogFile(), t)
+	// Goroutine execution order isn't guaranteed.
+	// If the object name in expected outcome doesn't align with the logs, swap
+	// the expected outcome objects and file names at positions 0 and 1.
+	if expectedOutcome[0].ObjectName != structuredReadLogs[0].ObjectName {
+		expectedOutcome[0], expectedOutcome[1] = expectedOutcome[1], expectedOutcome[0]
+		testFileNames[0], testFileNames[1] = testFileNames[1], testFileNames[0]
+	}
+	validate(expectedOutcome[0], structuredReadLogs[0], true, false, randomReadChunkCount, t)
+	validate(expectedOutcome[1], structuredReadLogs[1], true, false, randomReadChunkCount, t)
+	for i := 1; i < randomReadChunkCount; i++ {
+		ogletest.ExpectEq(false, structuredReadLogs[0].Chunks[i].IsSequential)
+		ogletest.ExpectEq(false, structuredReadLogs[0].Chunks[i].CacheHit)
```

Comment:
Cache hit false is expected right, as we are reading file on first time?

Reply:
Cache hit for follow up chunks is usually cache hit (even if we read the file first time). Here, cache hit false confirms that the file is not being cached.